institutions-ui/src/app/institutions/institutions.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { Institution } from "./institutions.models"; 
+
+const BASE_URL = '/api'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class InstitutionsService {
+  shortId(institutionId: String) {
+    // TODO this is a kludge
+    return institutionId.replace('https://osg-htc.org/iid/', '')
+  }
+
+  createInstitution(institution: Institution) {
+    return this.http.post(`${BASE_URL}/institutions`, institution)
+  }
+
+  updateInstitution(institutionId: string, institution: Institution) {
+    return this.http.put(`${BASE_URL}/institutions/${this.shortId(institutionId)}`, institution)
+  }
+
+  constructor(private http: HttpClient) {}
+
+  getInstitutions() {
+    return this.http.get<Institution[]>(`${BASE_URL}/institution_ids`)
+  }
+
+  getInstitutionDetails(institutionId: String) {
+    // TODO this is a kludge
```

Comment:
As someone not familiar with Angular, it's not immediately clear what the kludge is or what should be done to address it in the future